Datadog/Example/Debugging/DebugRUMViewController.swift

```diff
@@ -181,6 +180,51 @@ class DebugRUMViewController: UIViewController {
         simulatedViewControllers.append(viewController)
         sendErrorEventButton.disableFor(seconds: 0.5)
     }
+
+    // MARK: - Telemetry Events
+
+    @IBAction func didTapTelemetryEvent(_ sender: Any) {
+        guard let button = sender as? UIButton, let title = button.currentTitle else {
+            return
+        }
+        button.disableFor(seconds: 0.5)
```

Comment:
```suggestion
        button.isUserInteractionEnabled = false
        defer { button.isUserInteractionEnabled = true } 
```

Reply:
@mariedm 🤔 why? The `.disableFor(seconds: 0.5)` is pure utility, blocks the button for 0.5s and that's it.

Reply:
I imagine this is to prevent accidental gestures, such as double-tapping? My suggestion ensures the button is disabled while processing and re-enabled once the action is completed. But this is a nit and I'm fine with what you wrote.

Reply:
Actually, [`disableFor(seconds:)`](https://github.com/DataDog/dd-sdk-ios/blob/76fdab4a48c206e6d78f3e93972672c988f554fe/Datadog/Example/Utils/UIButton%2BDisabling.swift#L10-L27) does both: it prevents over-tapping and dims-out the button visually for given number of seconds.
```swift
self.isEnabled = false
self.backgroundColor = .systemGray
```